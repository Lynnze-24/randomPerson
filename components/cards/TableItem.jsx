import React from 'react'
import { View, Text,StyleSheet } from 'react-native'
import { Avatar } from 'react-native-paper'
import useMediaQueries from '../../hooks/MediaQueries'
import cusTheme from '../../theme/Theme'
import IconText from '../IconText'

const TableItem = ({name,avatar,email,phone,location}) => {
      const {isLessThanXLG,isLessThanXSM} = useMediaQueries()
    return (
        <View testID='tableItem' style={styles.container}>
            <View style={[styles.avatarCon,
                          isLessThanXLG && styles.avatarConXLG,
                          ]}>
                <Avatar.Image testID='avatar' size={60} source={avatar} />
                <Text testID='personName' style={[styles.text,isLessThanXLG && styles.textXLG]}>{name}</Text>
            </View>
            <View style={[styles.iconCon,isLessThanXSM && styles.iconConXSM]}>
                <IconText isTable={true} icon="email-outline" text={email} style={{flex:1.5}}/>
                <IconText isTable={true} icon="phone-outline" text={phone} />
                <IconText isTable={true} type="Ionicons" icon="location" text={location} style={{flex:1.5}}/>
            </View>
            
        </View>
    )
}

const styles = StyleSheet.create({
    container: {
      height:100,
      borderColor:cusTheme.lpr,
      borderWidth:2,
      borderTopWidth:0,
      justifyContent:'space-between',
      flexDirection:'row',
      alignItems:'center'
   },
   iconCon:{
       flex:'1',
    flexDirection:'row'
   },
   iconConXSM:{
        flexDirection:'column'
   },
   avatarCon:{
       marginLeft:20,
    flexDirection:'row',
    alignItems:'center'
   },
   avatarConXLG:{
       marginLeft:0,
    flexDirection:'column',
    justifyContent:'center'
   },
   text:{
       width:200,
       marginLeft:10,
       fontSize:13
   },
   textXLG:{
       textAlign:'center',
       marginLeft:0,
       width:150
    }
})

export default TableItem
