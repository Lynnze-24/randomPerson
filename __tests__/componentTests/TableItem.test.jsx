import {render } from "@testing-library/react-native";
import TableItem from "../../components/cards/TableItem.jsx"
import renderer from 'react-test-renderer';



it('matches snapShot', () => {
    const comp = renderer.create( <TableItem name="Aung"
                                    avatar={1} // mock
                                    email="gmail.com"
                                    phone="999"
                                    location="bago"
                                    />).toJSON();
    expect(comp).toMatchSnapshot();
});



it('should render component correctly',()=>{ 
    
    const {getByText,getByTestId}=render(
                        <TableItem name="Aung"
                                   avatar={1} // mock
                                   email="gmail.com"
                                   phone="999"
                                   location="bago"
                                    />
                        );
            getByTestId('avatar')
            getByText("Aung")
            getByText("gmail.com")
            getByText("999")
            getByText("bago")
        
})
