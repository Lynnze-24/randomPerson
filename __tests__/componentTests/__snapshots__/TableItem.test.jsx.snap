// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapShot 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "borderColor": "rgba(18,3,17,0.6)",
      "borderTopWidth": 0,
      "borderWidth": 2,
      "flexDirection": "row",
      "height": 100,
      "justifyContent": "space-between",
    }
  }
  testID="tableItem"
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "marginLeft": 20,
        },
        Object {
          "flexDirection": "column",
          "justifyContent": "center",
          "marginLeft": 0,
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "backgroundColor": "#6200ee",
            "borderRadius": 30,
            "height": 60,
            "width": 60,
          },
          undefined,
        ]
      }
      testID="avatar"
    >
      <Image
        source={1}
        style={
          Object {
            "borderRadius": 30,
            "height": 60,
            "width": 60,
          }
        }
      />
    </View>
    <Text
      style={
        Array [
          Object {
            "fontSize": 13,
            "marginLeft": 10,
            "width": 200,
          },
          Object {
            "marginLeft": 0,
            "textAlign": "center",
            "width": 150,
          },
        ]
      }
      testID="personName"
    >
      Aung
    </Text>
  </View>
  <View
    style={
      Array [
        Object {
          "flex": "1",
          "flexDirection": "row",
        },
        false,
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "flex": 1,
            "flexDirection": "row",
            "height": "100%",
          },
          Object {
            "flex": 1.5,
          },
        ]
      }
    >
      <Text />
      <Text
        style={
          Array [
            Object {
              "fontSize": 13,
              "marginLeft": 2,
              "textAlign": "left",
            },
            Object {
              "width": 100,
            },
            false,
          ]
        }
      >
        gmail.com
      </Text>
    </View>
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "flex": 1,
            "flexDirection": "row",
            "height": "100%",
          },
          undefined,
        ]
      }
    >
      <Text />
      <Text
        style={
          Array [
            Object {
              "fontSize": 13,
              "marginLeft": 2,
              "textAlign": "left",
            },
            Object {
              "width": 100,
            },
            false,
          ]
        }
      >
        999
      </Text>
    </View>
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "flex": 1,
            "flexDirection": "row",
            "height": "100%",
          },
          Object {
            "flex": 1.5,
          },
        ]
      }
    >
      <Text />
      <Text
        style={
          Array [
            Object {
              "fontSize": 13,
              "marginLeft": 2,
              "textAlign": "left",
            },
            Object {
              "width": 100,
            },
            false,
          ]
        }
      >
        bago
      </Text>
    </View>
  </View>
</View>
`;
